packages/textfield/test/textfield.test.ts

```diff
@@ -429,6 +431,49 @@ describe('Textfield', () => {
             : null;
         expect(input).to.not.be.null;
     });
+    it('valid - boundary-type assertions and title', async () => {
+        const el = await litFixture<Textfield>(
+            html`
+                <sp-textfield
+                    placeholder="Enter your number"
+                    pattern="^[\\d]+$"
+                    value="123"
+                ></sp-textfield>
+            `
+        );
+        await elementUpdated(el);
+
+        expect(el).to.not.equal(undefined);
+        const input = el.shadowRoot
+            ? el.shadowRoot.querySelector('#valid')
+            : null;
+        expect(input).to.not.be.null;
+        expect(el.shadowRoot.querySelector('input')).to.not.have.attribute(
```

Comment:
```suggestion
        expect(el.focusElement).to.not.have.attribute(
```
While we can reach into the shadow root, it's better not to when there are other paths to acquiring an element.